<!-- laser navigation testing-->
<launch>

<arg name="publish_map" default="true" />
<param name="use_sim_time" value="true" />

<include  file="$(find iri_laser_navigation)/launch/gazebo_erratic.launch"/>
<include  file="$(find erratic_navigation)/navigation_local/move_base_local.launch"/>

<node name="rviz" pkg="rviz" type="rviz"/>

<node name="iri_laser_icp"
      pkg="iri_laser_icp"
      type="iri_laser_icp"
      output="screen">
      <param name="~max_angular_correction_deg" value="30"/>
      <param name="~max_linear_correction" value="1"/>
      <param name="~max_iterations" value="30"/>
      <param name="~max_correspondence_dist" value="0.5"/>
      <param name="~clustering_threshold" value="20"/>
      <param name="~orientation_neighbourhood" value="3"/>
      <param name="~use_point_to_line_distance" value="1"/>
      <param name="~use_corr_tricks" value="1"/>
      <param name="~restart" value="1"/>
      <param name="~do_visibility_test" value="0"/>
      <param name="~do_compute_covariance" value="0"/>
</node>

<node name="iri_laser_localisation"
      pkg="iri_laser_localisation"
      type="iri_laser_localisation"
      output="screen">
      <remap from="~scan_sens" to="/base_scan/scan" />
      <remap from="~scan_ref" to="/iri_laser_navigation/scans_map" />
      <param name="~odom_frame" value="/odom" />
      <param name="~base_frame" value="/base_footprint" />
</node>

<node name="pwcs_to_odom"
      pkg="iri_msg_to_odom"
      type="posewcs_to_odom">
      <remap from="~pose" to="/iri_laser_localisation/pose"/>
      <remap from="~pose_ref" to="/iri_laser_navigation/pose_ref"/>
      </node>

<node name="iri_laser_navigation"
      pkg="iri_laser_navigation"
      type="iri_laser_navigation"
      output="screen">
      <remap from="~scan" to="/base_scan/scan"/>
      <param name="~bag_path"   value="$(find iri_laser_navigation)/launch/path_erratic.bag"/>
      <param name="~scan_topic" value="/base_scan/scan"/>
      <param name="~scan_path_frame_id" value="base_scan_link"/>
      <param name="~pose_topic" value="/odom"/>
      <param name="~goal_frame" value="/map"/>
      <param name="~publish_all" value="$(arg publish_map)"/>
</node>

</launch>
